utilities/plugins/openai_plugin.py

```diff
@@ -0,0 +1,225 @@
+        data = self._construct_request_data(endpoint, query, extra_param, streaming=True)
+        tokens = []
+        try:
+            url = f"{self.host}{endpoint}"
+            time.sleep(5)  # sometime while making multiple request it fails
+            response = requests.post(url, headers=headers, json=data, verify=False, stream=True)
+            LOGGER.info(response)
+            response.raise_for_status()
+            for line in response.iter_lines():
+                _, found, data = line.partition(b"data: ")
+                if found and data != b"[DONE]":
+                    message = json.loads(data)  # type: ignore
+                    token = self._parse_streaming_response(endpoint, message)
+                    tokens.append(token)
+        except (requests.exceptions.RequestException, json.JSONDecodeError):
+            LOGGER.exception("Streaming request error")
+
+        return "".join(tokens)
+
+    @staticmethod
+    def get_request_http(host: str, endpoint: str) -> Any:
+        """
+        Sends a HTTP GET request to the specified endpoint and returns the response data.
+
+        Args:
+            host (str): The base URL for the API.
+            endpoint (str): The API endpoint to send the request to.
+
+        Returns:
+            dict: The data from the response.
+
+        Raises:
+            requests.exceptions.RequestException: If there is a request error.
+            json.JSONDecodeError: If there is a JSON decoding error.
+        """
+        headers = {"Content-Type": "application/json"}
+        url = f"{host}{endpoint}"
+        try:
+            response = requests.get(url, headers=headers, verify=False)
+            LOGGER.info(response)
+            response.raise_for_status()
+            message = response.json()
+            data = message.get("data", [])
+            keys_to_remove = ["created", "id"]
+            if data:
+                data = OpenAIClient._remove_keys(data, keys_to_remove)
+            return data
+        except (requests.exceptions.RequestException, json.JSONDecodeError):
+            LOGGER.exception("Request error")
+
+    def _construct_request_data(
+        self,
+        endpoint: str,
+        query: Dict[str, Any],
+        extra_param: Optional[Dict[str, Any]] = None,
+        streaming: bool = False,
+    ) -> Dict[str, Any]:
+        """
+        Constructs the request data based on the endpoint and query parameters.
+
+        Args:
+            endpoint (str): The API endpoint to send the request to.
+            query (dict): The query parameters to include in the request.
+            extra_param (dict, optional): Additional parameters to include in the request.
+            streaming (bool, optional): If True, include streaming parameters. Defaults to False.
+
+        Returns:
+            Dict: The constructed request data.
+        """
+        data = {}
+        if "/v1/chat/completions" in endpoint:
```

Comment:
Save as constant and reuse. Maybe consider creating a nested class inside OpenAIClient to store the value of these endpoints

Reply:
I have create another constant.py file in the same directory to handle it